src/main/scala/components/memory/memory-noncombin-ports.scala

```diff
@@ -17,11 +17,22 @@ class OutstandingReq extends Bundle {
 
 /**
  * The instruction memory port. Since both the combinational and noncombinational instruction ports just issue
- * read requests in the same way both ports have the same implementation
+ * read requests in the same way both ports share the same implementation
  *
  * The I/O for this module is defined in [[IMemPortIO]].
  */
 class INonCombinMemPort extends ICombinMemPort {
+  // To fake combinational writes
+  val imemBusy  = RegInit (false.B)
+  imemBusy := DontCare
```

Comment:
This line worries me. Don't we want imemBusy to stay high or low until it changes? What does it mean to write a register with DontCare?

Reply:
This required a bit of testing but wiring a register with DontCare specifically makes it listen for any incoming signal on that cycle, and its necessary for the register to properly store values. Without the DontCare I believe as soon as it receives a high it will lock to being high. I'll elaborate with a timing chart similar to what I was reading while debugging it: 

Suppose the register starts with low and the write value alternates between low and high every cycle.
Without the DontCare line, the resulting register value will update like:
```
Cycle # | Register Val | Write Val
-------------------------------------
      1 |       0      |     0    
      2 |       0      |     1    
      3 |       1      |     0    
      4 |       1      |     1    
      5 |       1      |     0    
      6 |       1      |     1    
...
```
And if we initially write it with DontCare:
```
Cycle # | Register Val | Write Val
-------------------------------------
      1 |       0      |     0    
      2 |       0      |     1    
      3 |       1      |     0    
      4 |       0      |     1    
      5 |       1      |     0    
      6 |       0      |     1    
...
```
It appears to be a weird quirk with Chisel. All I know is that this line is required and removing it causes the register to act weird like in the first example.